File: PHP/project/tests/Unit/CH8/Sec2_3/LogAnalyzerTest.php
<?php

namespace Tests\Unit\CH8\Sec2_3;

use PHPUnit\Framework\TestCase;
use Src\CH8\Sec2_2\LogAnalyzer;

class LogAnalyzerTest extends TestCase
{
    /**
     * @var LogAnalyzer
     */
    private $LogAnalyzer;

    private $file_info;

    public function setUp(): void
    {
        parent::setUp();
        $this->LogAnalyzer = new LogAnalyzer();
        $this->LogAnalyzer->initialize();
        $this->file_info = 'c://someFile.txt';
    }

    /**
     * @test
     */
    public function IsValid_LengthBiggerThan8_IsFalse(): void
    {
        //Arrange
        $filename = '123456789';

        //Actual
        $actual = $this->LogAnalyzer->isValid($filename);

        //Assert
        $this->assertFalse($actual);
    }

    /**
     * @test
     */
    public function IsValid_LengthSmallerThan8_IsTrue(): void
    {    //Arrange
        $filename = '1234567';

        //Actual
        $actual = $this->LogAnalyzer->isValid($filename);

        //Assert
        $this->assertTrue($actual);
    }

    /**
     * @test
     */
    public function IsValid_BadFileInfoInput_ReturnsFalse(): void
    {    //Arrange
        $filename = $this->file_info;

        //Actual
        $actual = $this->LogAnalyzer->isValid($filename);

        //Assert
        $this->assertFalse($actual);
    }
}

File: C#/CKUT/CKUT.UnitTests/Ch8/8.2.3/LogAnalyzerTest.cs
using System.IO;
using CKUT.Core.Services.Ch8._8._2._2;
using NUnit.Framework;

namespace CKUT.UnitTests.Ch8._8._2._3
{
    [TestFixture]
    public class LogAnalyzerTest
    {
        private FileInfo _fileInfo;
        private LogAnalyzerInit _logan; 

        [SetUp]
        public void SetUp()
        {
            _logan = new LogAnalyzerInit();
            _logan.Initialize();
            _fileInfo = new FileInfo("c:\\someFile.txt");
        }
        [Test]
        public void IsValid_LengthBiggerThan8_IsFalse()
        {
            bool valid = _logan.IsValid("123456789");
            Assert.IsFalse(valid);
        }
        [Test]
        public void IsValid_BadFileInfoInput_returnFalse()
        {
            bool valid = _logan.IsValid(_fileInfo.Name);
            Assert.IsFalse(valid);
        }
        [Test]
        public void IsValid_LengthSmallerThan8_IsTrue()
        {
            bool valid = _logan.IsValid("1234567");
            Assert.IsTrue(valid);
        }
    }
}